includes/REST_API/Stories_Media_Controller.php

```diff
@@ -171,4 +175,46 @@ protected function prepare_items_query( $prepared_args = [], $request = null ) {
 	protected function get_media_types() {
 		return $this->get_allowed_mime_types();
 	}
+
+	/**
+	 * Filters the current query to hide all automatically extracted poster image attachments.
+	 *
+	 * Reduces unnecessary noise in the media library.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param \WP_Query $query WP_Query instance, passed by reference.
+	 *
+	 * @return void
+	 */
+	public function filter_poster_attachments( &$query ) {
+		$post_type = (array) $query->get( 'post_type' );
+
+		if ( ! in_array( 'any', $post_type, true ) && ! in_array( 'attachment', $post_type, true ) ) {
+			return;
+		}
+
+		$version = get_option( Database_Upgrader::OPTION, '0.0.0' );
+		if ( version_compare( '3.0.4', $version, '>=' ) ) {
```

Comment:
Once we merge #5635 (add term & migrate meta to it), we don't really need this check? Migration should have happened already? If not, when would we remove this check again?

Reply:
Sadly we don't know that. The migration runs on `admin_init`. What if the plugin is update automatically or by your hosting provider. We just never know when this database upgrade will happen. 

We could fix this issue, but hooking into [upgrader_post_install](https://github.com/WordPress/wordpress-develop/blob/e87d86b2ae1ffc48660a8fdced256ef3f94e8236/src/wp-admin/includes/class-wp-upgrader.php#L624) and force the database upgrade on plugin upgrade. But this is another ticket methinks.

Reply:
As soon as you open the editor, the upgrade will run. So the only scenario I see right now where this check would be needed:

1. Open editor
1. Plugin is updated in the background (e.g. by your hosting provider or in a separate tab)
1. Continue using editor -> upgrade never happens

The likelihood of that is rather small, and can be simply "fixed" by refreshing the editor.

Reply:
If you are happy then I will remove the checks and add tests to this PR.

Reply:
Yes please 👍